python/GafferUI/FileMenu.py

```diff
@@ -398,6 +433,19 @@ def __selectionAvailable( menu ) :
 	scriptWindow = menu.ancestor( GafferUI.ScriptWindow )
 	return True if scriptWindow.scriptNode().selection().size() else False
 
+def __backupAvailable( menu ) :
+
+	scriptWindow = menu.ancestor( GafferUI.ScriptWindow )
+	currentScript = scriptWindow.scriptNode()
+	applicationRoot = currentScript.ancestor( Gaffer.ApplicationRoot )
+
+	backups = GafferUI.Backups.acquire( applicationRoot, createIfNecessary = False )
+
+	if backups is not None and backups.recoveryFile( currentScript ) :
+		return True
+
+	return False;
```

Comment:
This fails when the File menu is opened from a new script that has never been saved :

```
  File "/home/john/dev/build/gaffer-1.2/python/GafferUI/FileMenu.py", line 444, in __backupAvailable
    if backups is not None and backups.recoveryFile( currentScript ) :
  File "/home/john/dev/build/gaffer-1.2/python/GafferUI/Backups.py", line 129, in recoveryFile
    backups = self.backups( scriptPath )
  File "/home/john/dev/build/gaffer-1.2/python/GafferUI/Backups.py", line 118, in backups
    paths = [ p for p in self.__potentialPaths( script ) if p.exists() ]
  File "/home/john/dev/build/gaffer-1.2/python/GafferUI/Backups.py", line 256, in __potentialPaths
    scriptPath = self.__scriptPath( script )
  File "/home/john/dev/build/gaffer-1.2/python/GafferUI/Backups.py", line 249, in __scriptPath
    assert( isinstance( script, str ) )
AssertionError
```

I think that is probably a bug in the Backups class itself, since the code here seems perfectly reasonable. But we'll need to fix it before we can merge. Would be good to add the fix in a separate commit, and include a test for it in BackupsTest.

Reply:
Done! Added in https://github.com/GafferHQ/gaffer/pull/5145/commits/271bc1583c47b62f94f977dd5889eb7343d317a2